File: SMXAMIOSTemplate/NotificationContentExtension/NotificationViewController.swift
//
//  NotificationViewController.swift
//  NotificationContentExtension
//
//  Created by Gilbert Schakal on 17/05/2018.
//  Copyright © 2018 Samy Ziat. All rights reserved.
//

import UIKit
import UserNotifications
import UserNotificationsUI

class NotificationViewController: UIViewController, UNNotificationContentExtension {

    @IBOutlet var label: UILabel?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any required interface initialization here.
    }
    
    func didReceive(_ notification: UNNotification) {
        self.label?.text = notification.request.content.body
        
        let url = "URL"
        let clientID = "ClientID"
        let privateKey = "privateKey"
        
        //  Create the SMManagerSetting instance
        let setting: SMManagerSetting = SMManagerSetting(url: url, clientID: clientID, privateKey: privateKey)
        //Start sdk from extension
        SMManager.sharedInstance().startExtension(with: setting)
        
        //sdk api to add buttons from banner
        SMManager.sharedInstance().didReceive(notification)
    }

}

File: NotificationContentExtension/NotificationViewController.cs
using System;
using Foundation;
using SelligentMobileiOS;
using UIKit;
using UserNotifications;
using UserNotificationsUI;


namespace NotificationContentExtension
{
    public partial class NotificationViewController : UIViewController, IUNNotificationContentExtension
    {
        protected NotificationViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            // Do any required interface initialization here.
        }

        public void DidReceiveNotification(UNNotification notification)
        {            
            var url = "https://mobile.slgnt.eu/MobilePush/api/";
            var clientID = "f06ff4cc-15c5-48b1-b405-d78dfc93f3de";
            var privateKey = "/5v3UKuFJ237PSOQy6BYtICR8+DaU/YWhjBz2JI3q8wrAIBCYt7eSWZith+sIFNWgQqdYdMx+EhYoZUa8NUsIw==";

            //  Create the SMManagerSetting instance
            var setting = SMManagerSetting.SettingWithUrl(url, clientID, privateKey);
            //Start sdk from extension
            SMManager.SharedInstance().StartExtensionWithSetting(setting);

            //sdk api to add buttons from banner
            SMManager.SharedInstance().DidReceiveNotification(notification);
            label.Text = notification.Request.Content.Body;
        }
    }
}
